src/sirius/filter.cc

```diff
@@ -40,16 +40,28 @@ namespace sirius {
 
 Image ZoomFilterImageToInputResolution(const Image& filter_image,
                                        const ZoomRatio& zoom_ratio);
+
 void NormalizeFilterImage(Image& filter_image, int oversampling);
-Image FrequencyShift(const Image& filter_image);
+
+Image FrequencyShift(const Image& filter_image, const Point& hot_point,
+                     float shift_col, float shift_row);
+
+Image CenterFilterImage(const Image& filter_image, const Point& hot_point);
 
 Filter Filter::Create(const std::string& image_path,
-                      const ZoomRatio& zoom_ratio, PaddingType padding_type,
-                      bool normalize) {
+                      const ZoomRatio& zoom_ratio, const Point& hot_point,
+                      PaddingType padding_type, bool filter_normalize) {
     // load image
     auto filter_image = gdal::LoadImage(image_path);
 
-    if (normalize) {
+    if (hot_point.x < -1 || hot_point.x >= filter_image.size.col ||
+        hot_point.y < -1 || hot_point.y >= filter_image.size.row) {
+        LOG("filter", error, "Invalid hot point with coordinates {}, {}",
+            hot_point.x, hot_point.y);
+        throw SiriusException("Invalid hot point");
```

Comment:
IMO, it'd best best to have a more complete error message and possibly to leave the log to the caller that catch the error.